Add release notes for Keycloak JS 26.2 (#39)
keycloak/keycloak-js: docs/release-notes/26_2_0.adoc

> Today marks a significant milestone in the evolution of Keycloak JS with the release of version `26.2.0`. This new version represents a shift in how the JavaScript adapter develops and evolves alongside the Keycloak ecosystem. Although this new version does not bring any functional changes to the adapter compared to previous versions, it does come with several organizational changes users should be made aware of.

Reviewer: ```suggestion
Today marks a significant milestone in the evolution of Keycloak JS with the release of version `26.2.0`. This new version represents a shift in how the JavaScript adapter develops and evolves alongside the {project_name} ecosystem. Although this new version introduces no functional changes to the adapter, it does include several organizational changes.
```

> The most notable change is that Keycloak JS now breaks free from the main Keycloak project's release cycle. As https://www.keycloak.org/2025/01/keycloak-2610-released#_dedicated_release_cycle_for_the_node_js_adapter_and_javascript_adapter[announced earlier this year], the JavaScript adapter will follow its own independent development path. The separation from the main project allows for more frequent releases of features, bugfixes, and improved responsiveness to community feedback. The JavaScript adapter will continue to be backwards compatible with all actively supported releases of the Keycloak server, and deviation from this will be considered a breaking change.

Reviewer: ```suggestion
The most notable change is that Keycloak JS now breaks free from the main Keycloak project's release cycle. As https://www.keycloak.org/2025/01/keycloak-2610-released#_dedicated_release_cycle_for_the_node_js_adapter_and_javascript_adapter[announced earlier this year], the JavaScript adapter will follow its own independent development path. The separation from the main project allows for more frequent releases of features, bug fixes, and improved responsiveness to community feedback. The JavaScript adapter will continue to be backwards compatible with all actively supported releases of the Keycloak server, and deviation from this will be considered a breaking change.
```

The choice to use a higher version than the main project itself was made intentionally in order to signal to users the departure from Keycloak's release cycle. We will however continue to commit to using https://semver.org/[Semantic Versioning], only bumping major versions if backwards incompatible changes are made, as is customary in the https://docs.npmjs.com/about-semantic-versioning[NPM ecosystem]. Maintenance updates will continue to land in the `26.1.x` series, as it is tied to the current stable release of the Keycloak server, but we encourage users to upgrade to new versions as needed.

> Another significant change is the relocation of the codebase to its https://github.com/keycloak/keycloak-js[own dedicated repository]. This structural adjustment isn't just administrative—it represents a strategic move toward better maintainability. By separating the JavaScript adapter from the main Keycloak repository, the development team gains greater flexibility in managing the codebase and processing community contributions. If you are looking to provide contributions, or are reporting issues, please redirect your efforts here.

Reviewer: ```suggestion
Another significant change is the relocation of the codebase to its https://github.com/keycloak/keycloak-js[own dedicated repository]. This structural adjustment is not just administrative—it represents a strategic move toward better maintainability. By separating the JavaScript adapter from the main Keycloak repository, the development team gains greater flexibility in managing the codebase and processing community contributions. If you are looking to provide contributions, or are reporting issues, please redirect your efforts here.
```

> Looking ahead, we will be focussing on what is next for Keycloak JS. When it was originally released, only a few OpenID Connect adapters existed for client-side JavaScript, so we needed to make our own adapter. However, this landscape looks very different now, and there are many mature solutions available. The code for Keycloak JS requires modernization and has become challenging to maintain due to the growing complexity. We will continue to evaluate if it makes sense to keep refactoring Keycloak JS, incorporate some mature third-party libraries we can collaborate on, or even replace it with a well-established community solution.

Reviewer: ```suggestion
Looking ahead, we will be focussing on what is next for Keycloak JS. When it was originally released, only a few OpenID Connect adapters existed for client-side JavaScript, so we needed to make our own adapters. However, this landscape looks very different now, and there are many mature solutions available. The code for Keycloak JS requires modernization and has become challenging to maintain due to the growing complexity. We will continue to evaluate if it makes sense to keep refactoring Keycloak JS, incorporate some mature third-party libraries we can collaborate on, or even replace it with a well-established community solution.
```

Author: We only made one client side adapter, so I think we need to keep this singular.

> Looking ahead, we will be focussing on what is next for Keycloak JS. When it was originally released there were very little OpenID Connect adapters for client-side JavaScript, and there was a need to make our own. However, this landscape looks very different now, and there are many mature solutions available. The code for Keycloak JS is also in need of modernization and has become hard to maintain over time due to the growing complexity. We will be evaluating whether it makes sense to keep refactoring Keycloak JS, incorporate some mature third-party libraries we can collaborate on, or even replace it with a well established community solution.

Reviewer: ```suggestion
Looking ahead, we will be focussing on what is next for Keycloak JS. When it was originally released, only a few OpenID Connect adapters existed for client-side JavaScript, so we needed to make our own adapters. However, this landscape looks very different now, and there are many mature solutions available. The code for Keycloak JS requires modernization and has become challenging to maintain due to the growing complexity. We will continue to evaluate if it makes sense to keep refactoring Keycloak JS, incorporate some mature third-party libraries we can collaborate on, or even replace it with a well-established community solution.
```
